foods/serializers.py

```diff
@@ -0,0 +1,60 @@
+from rest_framework import serializers
+from .models import Food, Category, Images
+from django.conf import settings
+
+
+class ImagesSerializer(serializers.ModelSerializer):
+    image_url = serializers.SerializerMethodField()
+
+    def get_image_url(self, obj):
+        return settings.HOST + obj.image_url.url
+
+    class Meta:
+        model = Images
+        fields = ['id', 'image_url']
+
+
+class CategorySerializer(serializers.ModelSerializer):
+    class Meta:
+        model: Category
+        fields: ['id', 'name', 'parent_id']
+
+
+class FoodSerializer(serializers.ModelSerializer):
+    images = ImagesSerializer(many=True)
+
+    class Meta:
+        model = Food
+        fields = ['id', 'name', 'category_id', 'price', 'discount', 'description', 'images']
+
+
+class PostOrPutFoodSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Food
+        fields = ['id', 'name', 'category_id', 'price', 'discount', 'description']
+    
+    def save_file(self, food):
+       for file in self.files:
+            Images.objects.create(image_url=file, food_id=food)
+
+    def save(self):
+        food = Food.objects.create(
+            name=self.validated_data.get('name'),
+            category_id=self.validated_data.get('category_id'),
+            price=self.validated_data.get('price'),
+            discount=self.validated_data.get('discount'),
+            description=self.validated_data.get('description'),
+        )
+        self.save_file(food)
+        return food
```

Comment:
```suggestion
    def save(self):
        food = FoodService.save_food(self.validated_data)
        return food
    
    class FoodService: # In another file
        @classmethod
        def save_food(cls, data):
            food = Food.objects.create(
            name=data.get('name'),
            category_id=data.get('category_id'),
            price=data.get('price'),
            discount=data.get('discount'),
            description=data.get('description'),
        )
        cls.save_file(food) # bring save_file there too
        return food
```
Create a `services` folder, create a `FoodService` class there. And put most of complex processes in that class. Example above

Reply:
Dạ